src/edgepi/gpio/edgepi_gpio_chip.py

```diff
@@ -83,12 +81,11 @@ def toggle_gpio_pin_state(self, pin_name: str = None):
         Return:
             N/A
         """
-        self.open_gpio(pin_num=self.__pin_name_dict[pin_name],
+        with self.open_gpio(pin_num=self.__pin_name_dict[pin_name],
                        pin_dir=self.gpiochip_pins_dict[pin_name].dir,
-                       pin_bias=self.gpiochip_pins_dict[pin_name].bias)
-        state = self.read_state()
-        if state:
-            self.write_state(False)
-        else:
-            self.write_state(True)
-        self.close_gpio()
+                       pin_bias=self.gpiochip_pins_dict[pin_name].bias):
+            state = self.read_state()
+            if state:
+                self.write_state(False)
+            else:
+                self.write_state(True)
```

Comment:
why not have one statement:

```
self.write_state(not state)
```

Reply:
[59fd062](https://github.com/EdgePi-Cloud/edgepi-python-sdk/pull/382/commits/59fd0624efdfb778e4b74de00b1df483aca8c43d)